pynicotine/privatechat.py

```diff
@@ -213,16 +213,24 @@ def _message_user(self, msg, queued_message=False):
         """Server code 22."""
 
         username = msg.user
+        message = msg.msg
 
         if not queued_message:
             log.add_chat(_("Private message from user '%(user)s': %(message)s"), {
                 "user": username,
-                "message": msg.msg
+                "message": message
             })
 
             core.send_message_to_server(slskmessages.MessageAcked(msg.msgid))
 
-        if username != "server":
+        if username == "server":
+            if message.startswith("The room you are trying to enter (") and ") " in message:
+                # Show fake chat room message if join wasn't successful
+                msg.user = None
+                room = message[message.find(" (") + 2:message.rfind(") ")]
```

Comment:
The position index of `" ("` is known within this explicit string, so there is no need to `find` it.